src/printer.js

```diff
@@ -2106,11 +2122,29 @@ function printArgumentsList(path, options, print) {
     const shouldBreak = shouldGroupFirst
       ? printed.slice(1).some(willBreak)
       : printed.slice(0, -1).some(willBreak);
+
+    let printedLastArgExpanded;
+    if (shouldGroupFirst) {
+      printedLastArgExpanded = printed;
+    } else {
+      // We want to print the last argument with a special flag
+      let i = 0;
+      path.each(function(argPath) {
+        if (i++ === args.length - 1) {
+          printedLastArgExpanded = printed
+            .slice(0, -1)
+            .concat(path.call(
+              print.withArgs({expandLastArg: true})
```

Comment:
`print` is not `genericPrint`. This is pretty confusing, and we should probably make this clearer somehow. `print` is the function passed in, which is `printGenerically`, which you adds the `args` argument below. The function in turn calls `genericPrint` with all this arguments: `p, options, printGenerically, args`. So the options that I passed in my snippet is the `args` that is being passed as the fourth argument to `genericPrint`.

I think that's right?